processors/src/NewVertexAnaProcessor.cxx

```diff
@@ -450,11 +445,15 @@ bool NewVertexAnaProcessor::process(IEvent* ievent) {
 
         if (smearingTool_ and !isData_) {
             double unsmeared_prod = ele_trk.getP()*pos_trk.getP();
-            double ele_smf = smearingTool_->updateWithSmearP(ele_trk, ele);
-            double pos_smf = smearingTool_->updateWithSmearP(pos_trk, pos);
+            double ele_smf = smearingTool_->updateWithSmearP(ele_trk);
+            double pos_smf = smearingTool_->updateWithSmearP(pos_trk);
             smearingTool_->updateVertexWithSmearP(vtx, ele_smf, pos_smf);
         }
 
+        //After all modifications to the track are made, update the memory address that the particle gets the track from
+        //with the modified track, that way changes here persist throughout this processor
```

Comment:
Just to be clear, the Particle holds a **copy** of the Track and CalCluster associated with it.

This is important to emphasize since the `ele_trk` and `ele.getTrack()` could evolve differently if only one is edited.